packages/gatsby-link/index.d.ts

```diff
@@ -1,11 +1,21 @@
-import * as React from "react";
+import React from "react"
+import { LinkProps } from "reach__router"
```

Comment:
Shouldn't it be like this?

```ts
import { LinkProps } from "@reach/router";
```

The import above will resolve to `@types/reach__router` anyway. Also it's a good idea to add `@types/reach__router` as a devDependency on `gatsby-link`

Reply:
Also, `import React from "react"` might fail on projects with `allowSyntheticDefaultImports` to `false` on their tsconfig, so it's best to keep it as it is (with `import *`).